velox/functions/prestosql/aggregates/ArbitraryAggregate.cpp

```diff
@@ -258,23 +320,46 @@ class NonNumericArbitrary : public exec::Aggregate {
   }
 
  protected:
-  // Initialize each group, we will not use the null flags because
-  // SingleValueAccumulator has its own flag.
+  // Initialize each group, we will not use the null flags because the
+  // accumulator has its own flag.
   void initializeNewGroupsInternal(
       char** groups,
       folly::Range<const vector_size_t*> indices) override {
     for (auto i : indices) {
-      new (groups[i] + offset_) SingleValueAccumulator();
+      if (clusteredInput_) {
+        new (groups[i] + offset_) ClusteredNonNumericAccumulator();
+      } else {
+        new (groups[i] + offset_) SingleValueAccumulator();
+      }
     }
   }
 
   void destroyInternal(folly::Range<char**> groups) override {
     for (auto group : groups) {
       if (isInitialized(group)) {
-        value<SingleValueAccumulator>(group)->destroy(allocator_);
+        if (clusteredInput_) {
+          auto* accumulator = value<ClusteredNonNumericAccumulator>(group);
+          std::destroy_at(accumulator);
+        } else {
+          auto* accumulator = value<SingleValueAccumulator>(group);
+          accumulator->destroy(allocator_);
+        }
       }
     }
   }
+
+  void addOrMergeCopyRange(const BaseVector::CopyRange& range) {
+    if (!copyRanges_.empty() && copyRanges_.back().mergeable(range)) {
+      ++copyRanges_.back().count;
+    } else {
+      copyRanges_.push_back(range);
+    }
+  }
+
+ private:
+  DecodedVector decoded_;
+  std::vector<BaseVector::CopyRange> copyRanges_;
+  std::vector<ClusteredNonNumericAccumulator*> accumulators_;
```

Comment:
Would it be possible to document these variables? Where do we use accumulators_?